src/mdb/mdb_shell.py

```diff
@@ -203,24 +206,32 @@ def do_execute(self, line: str) -> None:
 
             (mdb) execute test.mdb
         """
-
-        def strip_comments(text: str) -> str | None:
-            if re.match(r"^\s*#.*", text):
-                return None
-            return text
-
         file = line
         try:
             with open(file) as infile:
-                commands = infile.read().splitlines()
-                # strip comments from list of commands (lines starting with `#`)
-                commands = list(filter(strip_comments, commands))
-                self.cmdqueue.extend(commands)
+                contents = infile.read()
+            self.execute_script(contents, queue=True)
         except FileNotFoundError:
             print(
                 f"File [{file}] not found. Please check the file exists and try again."
             )
 
+    def execute_script(self, script: str, queue: bool = False) -> None:
+        def strip_comments(text: str) -> str | None:
+            if re.match(r"^\s*#.*", text):
+                return None
+            return text
+
+        commands = script.splitlines()
+        # strip comments from list of commands (lines starting with `#`)
+        commands = list(filter(strip_comments, commands))
+
+        if queue:
+            self.cmdqueue.extend(commands)
+        else:
+            for command in commands:
+                self.onecmd(self.precmd(command))
```

Comment:
Looking back at the commits, I added this so that you could give the shell a script to execute, and if you don't specify `queue=True` it would immediately fire them off one by one. The commit message I wrote in the change states it was to anticipate automating the test suite better. In the following commit (https://github.com/TomMelt/mdb/pull/47/commits/a01d1c9934005d35f52e2980a3c77bfe7eb26a33) I then used it implicitly with`queue=False` to automate the test script running.

I can't fully remember why these situations are different, i.e. why the test suite couldn't just queue commands, but looking at this again, I agree and don't see why we would ever do `onecmd` over `cmdqueue.extend`.

Reply:
Thanks for the explanation. I think it makes sense to take it out for now. Then I can also remove my weird hack thing too.